server/server_media.go

```diff
@@ -14,32 +14,38 @@ import (
 	"github.com/status-im/status-go/signal"
 )
 
-var (
-	// UseHTTP controls whether the media server uses HTTP instead of HTTPS.
-	// Set to true to avoid TLS certificate issues with react-native-fast-image
-	// on Android, which has limitations with dynamic certificate updates.
-	// Pls check doc/use-status-backend-server.md in status-mobile for more details
-	UseHTTP = false
-)
-
 type MediaServer struct {
 	Server
 
 	db              *sql.DB
 	downloader      *ipfs.Downloader
 	multiaccountsDB *multiaccounts.Database
 	walletDB        *sql.DB
+
+	// UseHTTP controls whether the media server uses HTTP instead of HTTPS.
+	// Set to true to avoid TLS certificate issues with react-native-fast-image
+	// on Android, which has limitations with dynamic certificate updates.
+	// Pls check doc/use-status-backend-server.md in status-mobile for more details
+	useHTTP bool
+}
+
+func initMediaCertificate(useHTTP bool) (*tls.Certificate, error) {
+	if useHTTP {
+		return nil, nil
+	}
+
+	cert, _, err := generateMediaTLSCert()
+	if err != nil {
+		return nil, err
+	}
+	return cert, nil
 }
 
 // NewMediaServer returns a *MediaServer
-func NewMediaServer(db *sql.DB, downloader *ipfs.Downloader, multiaccountsDB *multiaccounts.Database, walletDB *sql.DB) (*MediaServer, error) {
-	var cert *tls.Certificate
-	if !UseHTTP {
-		err := generateMediaTLSCert()
-		if err != nil {
-			return nil, err
-		}
-		cert = globalMediaCertificate
+func NewMediaServer(db *sql.DB, downloader *ipfs.Downloader, multiaccountsDB *multiaccounts.Database, walletDB *sql.DB, useHTTP bool) (*MediaServer, error) {
```

Comment:
It's too many arguments here already. Can you please make it `[]Options`?
Also, if you make it `opts ...Option`, you'd not need to change all usages of `NewMediaServer`, only the ones with non-default behaviour.